biz.aQute.repository/src/aQute/bnd/deployer/repository/ProgressWrappingStream.java

```diff
@@ -12,8 +15,60 @@
 	private Task		task;
 
 	public ProgressWrappingStream(InputStream delegate, String name, int size, ProgressPlugin progressPlugin) {
+		this(delegate, name, size, Collections.singletonList(progressPlugin));
+	}
+
+	public ProgressWrappingStream(InputStream delegate, String name, int size, List<ProgressPlugin> progressPlugins) {
 		this.delegate = delegate;
-		task = progressPlugin.startTask(name, size);
+
+		if (progressPlugins != null && progressPlugins.size() > 1) {
+			final List<ProgressPlugin.Task> multiplexedTasks = new ArrayList<>();
+
+			for (ProgressPlugin progressPlugin : progressPlugins) {
+				multiplexedTasks.add(progressPlugin.startTask(name, size));
+			}
+
+			task = new ProgressPlugin.Task() {
+				@Override
+				public void worked(int units) {
+					for (ProgressPlugin.Task multiplexedTask : multiplexedTasks) {
```

Comment:
To be nitpicky, the loop variable should be called `task` so it is the same as the other method's loops.